nevergrad/functions/control/core.py

```diff
@@ -40,19 +40,22 @@ class BaseFunction(ExperimentFunction):
     """
 
     def __init__(self, num_rollouts: int, random_state: tp.Optional[int] = None) -> None:
-        super().__init__(self._simulate, p.Array(shape=self.policy_dim))
+        if self.policy_2ndlayer_dim is not None:
+            super().__init__(self._simulate, np.p.Instrumentation(p.Array(shape=self.policy_dim), p.Array(shape=self.policy_2ndlayer_dim).set_mutation(sigma=1.e-4)))
+        else:
+            super().__init__(self._simulate, p.Array(shape=self.policy_dim))
```

Comment:
avoid calling super in a if statement, it's always better to uncouple everything that can be uncoupled. In this case create the parametrization in the if, then call init afterwards

Reply:
Done.